File: src/main/kotlin/org/daonators/client/request/RecoverPasswordByMailRequest.kt
package org.daonators.client.request

import br.com.simpli.model.LanguageHolder
import br.com.simpli.tools.Validator
import javax.ws.rs.BadRequestException

/**
 * Recover Password By Mail Request Model
 * @author Simpli CLI generator
 */
class RecoverPasswordByMailRequest(var email: String?) {
    fun validate(lang: LanguageHolder) {
        if (email.isNullOrEmpty()) {
            throw BadRequestException(lang.cannotBeNull("Email"))
        }

        if (!Validator.isEmail(email)) {
            throw BadRequestException(lang.isNotAValidEmail("Email"))
        }
    }
}

File: src/main/webapp/src/model/request/RecoverPasswordByMailRequest.ts
/**
 * RecoverPasswordByMailRequest
 * @author Simpli CLI generator
 */
import {$} from '@/facade'
import {Request} from '@simpli/serialized-request'

/* TODO: review generated class */
export class RecoverPasswordByMailRequest {
  email: string | null = null

  /**
   * Sends an email requesting to change the password
   */
  async recoverPasswordByMail() {
    return await Request.put(`/client/auth/password`, this)
      .name('recoverPasswordByMail')
      .asNumber()
      .getData()
  }
}
